vespa/application.py

```diff
@@ -582,6 +581,157 @@ def _handle_result_callback(
+        :param schema: The Vespa schema name that we are sending data to.
+        :param namespace: The Vespa document id namespace. If no namespace is provided the schema is used.
+        :param callback: A callback function to be called on each result. Signature `callback(response:VespaResponse, id:str)`
+        :param operation_type: The operation to perform. Default to `feed`. Valid are `feed`, `update` or `delete`.
+        :param max_queue_size: The maximum number of tasks waiting to be processed. Useful to limit memory usage. Default is 1000.
+        :param max_workers: Maximum number of concurrent requests to have in-flight, bound by an asyncio.Semaphore, that needs to be acquired by a submit task. Increase if the server is scaled to handle more requests.
+        :param max_connections: The maximum number of connections passed to httpx.AsyncClient to the Vespa endpoint. As HTTP/2 is used, only one connection is needed.
+        :param kwargs: Additional parameters are passed to the respective operation type specific :func:`_data_point`.
+        """
+
+        if operation_type not in ["feed", "update", "delete"]:
+            raise ValueError(
+                "Invalid operation type. Valid are `feed`, `update` or `delete`."
+            )
+
+        if namespace is None:
+            namespace = schema
+        if not schema:
+            try:
+                schema = self._infer_schema_name()
+            except ValueError:
+                raise ValueError(
+                    "Not possible to infer schema name. Specify schema parameter."
+                )
+
+        async def handle_result(task: asyncio.Task, id: str):
+            # Wrapper around the task to handle exceptions and call the user callback
+            try:
+                response = await task
+            except Exception as e:
+                response = VespaResponse(
+                    status_code=599,
+                    json={
+                        "Exception": str(e),
+                        "id": id,
+                        "message": "Exception during feed_data_point",
+                    },
+                    url="n/a",
+                    operation_type=operation_type,
+                )
+            if callback is not None:
+                try:
+                    callback(response, id)
+                except Exception as e:
+                    print(f"Exception in user callback for id {id}", file=sys.stderr)
+                    traceback.print_exception(
+                        type(e), e, e.__traceback__, file=sys.stderr
+                    )
+
+        # Wrapping in async function to be able to use asyncio.run, and avoid that the feed_async_iterable have to be async
+        async def run():
+            async with self.asyncio(connections=max_connections) as async_session:
+                semaphore = asyncio.Semaphore(max_workers)
+                tasks = []
+                for doc in iter:
```

Comment:
Feeding docs one by one seems not very efficient.
Is there any way to feed them in batches?

Reply:
Since this is async, all the tasks will be created immediately here, without waiting for the tasks execution, and the tasks will be executed concurrently (as long as a semaphore is available). 
To limit memory usage, there is a batching of tasks instead, to avoid memory issues, if the iterable contains more documents than fits into memory, for instance. 
Did that make it any clearer?

Reply:
Still every document is still one request that needs to be processed on the server.
A lot of network and processing overhead.
Usually any data store that needs to write some non-tiny amount of data prefers batch feeding.

Reply:
No batch feeding.